### PR TITLE
Add libpqxx/7.6.0

### PR DESCRIPTION
**Pull request by tt4g**
Specify library name and version:  **libpqxx/7.6.0**

New minor version released: https://github.com/jtv/libpqxx/releases/tag/7.6.0

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by tt4g** [comment]
I created pull request jtv/libpqxx#465 and confirmed that the build succeeds on Docker image `conanio/gcc7` with this patch.

Please wait for the pull request to be merged.

**Comment by tt4g** [comment]
OK.
Since libpqxx 7.5.0 no longer supports gcc 7 (see this comment https://github.com/jtv/libpqxx/issues/464#issuecomment-888358662).

My last commit was to drop support for gcc 7.

File: recipes/libpqxx/all/conanfile.py
Comment (tt4g) [reply recipes/libpqxx/all/conanfile.py:46]: Fixed f837c4ca38de78cf32ea64ef001dd0d325f93978

File: recipes/libpqxx/all/conanfile.py
Comment (tt4g) [reply recipes/libpqxx/all/conanfile.py]: Fixed 46a35a79a06cfdfc9b576d5d687a778d36730570

File: recipes/libpqxx/all/conanfile.py
Comment (SSE4) [inline recipes/libpqxx/all/conanfile.py:86]: that's interesting, isn't it set by conan already?

**Comment by jgsogo** [comment]
Hi! Your PR (https://github.com/jtv/libpqxx/pull/468) was merged, do you think we can/should wait for version 7.6.1?

File: recipes/libpqxx/all/conanfile.py
Comment (tt4g) [reply recipes/libpqxx/all/conanfile.py:86]: @SSE I thought so too, but the CI failed to build apple-clang 11 (https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/6566/9/libpqxx/7.6.0//summary.json).
Several recipes had this CMake variable set, so I did the same for this recipe and the build succeeded.

**Comment by tt4g** [comment]
@jgsogo As far as I know, there is no release scheduling for the next version.
Since it may take more than a few months before the next release, I think it is desirable to make this release with the new C++20 features available from conan as soon as possible.

File: recipes/libpqxx/all/conanfile.py
Comment (jgsogo) [inline recipes/libpqxx/all/conanfile.py:48]: Previously, versions `7.5.x` were being built with Visual Studio 15 (and GCC 7 and Clang 6), right? With this change we are discarding those versions...

Comment (tt4g) [reply recipes/libpqxx/all/conanfile.py:48]: Yes. libpqxx 7.6.0 no longer supports these versions.

Visual Studio 15 (2017) cannot build because libpqxx 7.6.0 require `[[deprecated]]` attribute support.
However, Visual Studio 15 support is not complete and the build will fail.
https://github.com/jtv/libpqxx/blob/7.6.0/NEWS#L17

MSVC issues:
  * https://developercommunity.visualstudio.com/t/compilation-error-on-deprecated-template-construct/139521
  * https://developercommunity.visualstudio.com/t/deprecated-attribute-with-template-member-function/421223

gcc 6 was already unsupported in libpqxx version 7.5.0 (it could be built, but there was no guarantee that it would work).
https://github.com/jtv/libpqxx/blob/7.6.0/NEWS#L34

I couldn't get clang 6 to work, but it may be possible to build it with the PR (jtv/libpqxx#468) patch.
I'll check it once and see if it works.

Comment (tt4g) [reply recipes/libpqxx/all/conanfile.py:48]: @jgsogo clang 6 was able to build libpqxx 7.6.0 with patch `fix-inline-constexpr-local-variable-problem-7.6.0.patch`.
Resumed support for clang 6 in the last commit (e695f6181e467fce88ba8d2b442154872f317d0b).

Comment (jgsogo) [reply recipes/libpqxx/all/conanfile.py:48]: I think I didn't manage to explain myself. I meant it should be:

```
    lib_version_7_6_0_or_above = Version(self.version) >= "7.6.0"
    minimum_compiler_version = {
            "Visual Studio": "16" if lib_version_7_6_0_or_above else "15",
```

It is the new version `7.6.0` the one that requires new compiler versions. Old versions (7.5.x series) were _working_ with those compiler versions.

Comment (tt4g) [reply recipes/libpqxx/all/conanfile.py:48]: > Old versions (7.5.x series) were working with those compiler versions.

The previous version used macros to work around the MSVC 15 bug (jtv/libpqxx#406).
However, in order to continue to provide this workaround using `__declspec(deprecated)`, there was the problem of having to define multiple macros when other C++ standard attributes and `__declspec` wanted to be used for functions or classes.

In this version, the maintainer made the decision to remove this workaround.

Comment (jgsogo) [reply recipes/libpqxx/all/conanfile.py:48]: Yes, but the changes in this PR are dropping support for some compiler versions in `libpqxx/7.5.x`, see the output (https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/6566/16-production/libpqxx/7.5.2//summary.json):

image

and they were available before:

`conan search libpqxx/7.5.2@ -r conancenter --table=output.html`
image


I'm just saying that **the condition should be _version 7.6.0 or above_** (not 7.5.0 like it is now in this PR).

Comment (tt4g) [reply recipes/libpqxx/all/conanfile.py]: @jgsogo Oops, that was an oversight on my part. Thanks for bringing it to my attention!

Fixed the conditional expression in the last commit (09d2f88f12527be2d21474be2a20cee0726819c8).

**Review by jgsogo — APPROVED** [review]
LGTM.

---

> gcc 6 was already unsupported in libpqxx version 7.5.0 (it could be built, but there was no guarantee that it would work).
> https://github.com/jtv/libpqxx/blob/7.6.0/NEWS#L34

👍

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**
